File: WinForms/EqDemoWinFormsNet4/MainForm.cs
using System;
using System.Threading;
using System.Windows.Forms;

namespace EqDemo
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs ev)
        {
            button1.Invoke((MethodInvoker)delegate {
                button1.Text = "Loading...";
                button1.Enabled = false;
            });

            var eqForm = new EasyQueryForm();
            eqForm.FormClosed += (s, e) => Application.Exit();
            Hide();
            eqForm.Show();
        }
    }
}

File: WinForms/EqDemoWinFormsNet4VB/MainForm.vb
Imports System
Imports System.Threading
Imports System.Windows.Forms

Partial Public Class MainForm
    Inherits Form

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub button1_Click(ByVal sender As Object, ByVal ev As EventArgs)
        button1.Invoke(CType(Function()
                                 button1.Text = "Loading..."
                                 button1.Enabled = False
                             End Function, MethodInvoker))
        Dim eqForm = New EasyQueryForm()
        AddHandler eqForm.FormClosed, (Function(s, e)
                                           Application.Exit()
                                       End Function)
        Hide()
        eqForm.Show()
    End Sub
End Class
